### ISSUE TITLE
Button - update console error for accessible name and move aria-expanded location

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#11871]
**Describe the enhancement or change**
Followup to some comments left in https://github.com/patternfly/patternfly-react/pull/11861/

- The console error with the `isSettings || isHamburger || isFavorite` conditional for the accessible name (line 164 in that PR) should be updated to remove that conditional. We should be throwing an error for any button that doesn't have children, aria-label, or aria-labelledby
- The `aria-expanded` attribute (line 239) should be moved below the `{...props}` so that it won't be overridden by a manual `aria-expanded` being passed in
- Several tests related to the above points will need to be updated in some fashion as well

**Is this request originating from a Red Hat product team? If so, which ones and is there any sort of deadline for this enhancement?**

**Any other information?**
